pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/renderer/TextValueProvider.java

```diff
@@ -1,101 +1,224 @@
+                        .collect(Collectors.joining(", "));
+            }
+            return val.toString();
+        }).collect(Collectors.toList());
+    }
+
+    /** handles reading values from things other than questions. including basic properties and array properties
+     * it falls back to reading the dynamic fields if an object is specified that doesn't exist on the
+     * is found
+     */
+    protected Object getValueFromMap(Map<String, Object> moduleMap, FilterExportConfig filterConfig) {
+        Map<String, Object> targetMap = null;
+        String objectName = filterConfig.getColumn().getObject();
+        String fieldName = filterConfig.getColumn().getName();
+        if (objectName != null) {
+            Object formObject = moduleMap.get(objectName);
+            if (formObject instanceof List) {
+                // this is a list, like "testResult" off of kitRequestShipping
+                // transform an array of maps into a map of arrays
+                Map<String, Object> answerMap = new HashMap();
+                final String finalFieldName = fieldName;
+                List<Object> answersArray = ((List<?>) formObject).stream().map(arrValue -> {
+                    if (arrValue instanceof Map) {
+                        return ((Map<String, Object>) arrValue).get(finalFieldName);
+                    }
+                    return StringUtils.EMPTY;
+                }).collect(Collectors.toList());
+                answerMap.put(fieldName, answersArray);
+                targetMap = answerMap;
+
+            } else if (formObject instanceof Map) {
+                targetMap = (Map<String, Object>) formObject;
+            }
         }
+        if ((targetMap == null && objectName != null) || ESObjectConstants.ADDITIONAL_VALUE.equals(filterConfig.getType())) {
+            // try dynamic fields
+            Map<String, Object> dynamicFieldMap = (Map<String, Object>) moduleMap.get(ESObjectConstants.DYNAMIC_FIELDS);
+            if (dynamicFieldMap != null) {
+                // the field name may be given in either snake case or camelcase, but in the dynamic field map, it is camel case
+                String camelCasedFieldName = fieldName;
+                if (Character.isUpperCase(camelCasedFieldName.charAt(0))) {
+                    camelCasedFieldName = Arrays.stream(fieldName.split("_"))
+                            .map(word -> StringUtils.capitalize(StringUtils.toRootLowerCase(word))).collect(Collectors.joining());
+                    camelCasedFieldName = StringUtils.uncapitalize(camelCasedFieldName);
+                }
 
-        return answer;
+                if (dynamicFieldMap.get(camelCasedFieldName) != null) {
+                    targetMap = dynamicFieldMap;
+                    fieldName = camelCasedFieldName;
+                }
+            }
+        }
+        // if this isn't any of the special cases above, just use the passed-in map
+        if (targetMap == null) {
```

Comment:
I would introduce command design pattern here with its factory class, this would help to separate logics in its own classes, like `ListValueHandler`, `DynamicFieldValueHandler`, `BasicValueHandler`

Reply:
in my opinion it is not good idea to wait it may become complex or not, having several if statements to make different things already means that it is enough complex to separate them into different classes.